packages/expo-updates/ios/EXUpdates/Database/EXUpdatesDatabase.m

```diff
@@ -449,6 +452,97 @@ - (nullable EXUpdatesAsset *)assetWithKey:(NSString *)key error:(NSError ** _Nul
+  NSArray<NSDictionary *> *rows = [self _executeSql:sql withArgs:@[key, scopeKey] error:error];
+  if (rows && [rows count]) {
+    id value = rows[0][@"value"];
+    if (value && [value isKindOfClass:[NSString class]]) {
+      NSDictionary *jsonObject = [NSJSONSerialization JSONObjectWithData:[(NSString *)value dataUsingEncoding:NSUTF8StringEncoding] options:kNilOptions error:error];
+      if (!*error && jsonObject && [jsonObject isKindOfClass:[NSDictionary class]]) {
+        return jsonObject;
+      }
+    }
+  }
+  return nil;
+}
+
+- (BOOL)_setJsonData:(NSDictionary *)data withKey:(NSString *)key scopeKey:(NSString *)scopeKey isInTransaction:(BOOL)isInTransaction error:(NSError ** _Nullable)error
+{
+  if (!isInTransaction) {
+    sqlite3_exec(_db, "BEGIN;", NULL, NULL, NULL);
+  }
+  NSString * const deleteSql = @"DELETE FROM json_data WHERE \"key\" = ?1 AND \"scope_key\" = ?2";
+  if ([self _executeSql:deleteSql withArgs:@[key, scopeKey] error:error] == nil) {
+    if (!isInTransaction) {
+      sqlite3_exec(_db, "ROLLBACK;", NULL, NULL, NULL);
+    }
+    return NO;
+  }
+
+  NSString * const insertSql = @"INSERT INTO json_data (\"key\", \"value\", \"last_updated\", \"scope_key\") VALUES (?1, ?2, ?3, ?4);";
+  if ([self _executeSql:insertSql withArgs:@[key, data, @(NSDate.date.timeIntervalSince1970 * 1000), scopeKey] error:error] == nil) {
+    if (!isInTransaction) {
+      sqlite3_exec(_db, "ROLLBACK;", NULL, NULL, NULL);
+    }
+    return NO;
+  }
+  if (!isInTransaction) {
+    sqlite3_exec(_db, "COMMIT;", NULL, NULL, NULL);
+  }
+
+  return YES;
+}
+
+- (nullable NSDictionary *)serverDefinedHeadersWithScopeKey:(NSString *)scopeKey error:(NSError ** _Nullable)error
+{
+  return [self _jsonDataWithKey:EXUpdatesDatabaseServerDefinedHeadersKey scopeKey:scopeKey error:error];
+}
+
+- (nullable NSDictionary *)manifestFiltersWithScopeKey:(NSString *)scopeKey error:(NSError ** _Nullable)error
+{
+  return [self _jsonDataWithKey:EXUpdatesDatabaseManifestFiltersKey scopeKey:scopeKey error:error];
+}
+
+- (void)setServerDefinedHeaders:(NSDictionary *)serverDefinedHeaders withScopeKey:(NSString *)scopeKey error:(NSError ** _Nullable)error
+{
+  [self _setJsonData:serverDefinedHeaders withKey:EXUpdatesDatabaseServerDefinedHeadersKey scopeKey:scopeKey isInTransaction:NO error:error];
+}
+
+- (void)setManifestFilters:(NSDictionary *)manifestFilters withScopeKey:(NSString *)scopeKey error:(NSError ** _Nullable)error
+{
+  [self _setJsonData:manifestFilters withKey:EXUpdatesDatabaseManifestFiltersKey scopeKey:scopeKey isInTransaction:NO error:error];
+}
+
+- (void)setServerDataWithManifest:(EXUpdatesUpdate *)updateManifest error:(NSError ** _Nullable)error
+{
+  sqlite3_exec(_db, "BEGIN;", NULL, NULL, NULL);
+  if (updateManifest.serverDefinedHeaders) {
+    if ([self _setJsonData:updateManifest.serverDefinedHeaders
+                   withKey:EXUpdatesDatabaseServerDefinedHeadersKey
+                  scopeKey:updateManifest.scopeKey
```

Comment:
So the `scopeKey` can change if the project is transferred correct? And this is used to filter only the current project owner's metadata?

Reply:
Yeah -- this is basically only used in the Expo Go clients, where we have data for multiple different projects stored in the same SQLite db.

Reply:
Realized I didn't address your first question -- the idea is that the `scopeKey` is something that will NOT change and will remain constant across project transfers.